src/vignette/util/image_response.clj

```diff
@@ -38,13 +38,14 @@
 
 (defn create-image-response
   ([image image-map]
-   (-> (response (->response-object image))
-       (header "Content-Type" (content-type image))
-       (header "Content-Length" (content-length image))
-       (header "ETag" (etag image))
-       (header "X-Thumbnailer" "Vignette")
-       (add-content-disposition-header image-map)
-       (add-surrogate-header image-map)))
+
+   (as-> (response (->response-object image)) resp
+         (if-let [ct (content-type image)] (header resp "Content-Type" ct) resp)
+         (if-let [cl (content-length image)] (header resp "Content-Length" cl) resp)
+         (if-let [etag (etag image)] (header resp "ETag" etag) resp)
```

Comment:
Minor: shouldn't all of the image maps have these headers? Also the inlined `if-let`s are hard to follow. I think I would prefer a `header` wrapper than was a noop if the input was nil so the readability could be maintained.

Something like
```clojure
(defn if-header
   [resp key val]
   (if val
       (header resp key val)
       resp))
```